src/OrchardCore.Modules/OrchardCore.Navigation/Startup.cs

```diff
@@ -6,13 +7,21 @@ namespace OrchardCore.Navigation
 {
     public class Startup : StartupBase
     {
+        private readonly IConfiguration _configuration;
+
+        public Startup(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
         public override void ConfigureServices(IServiceCollection services)
         {
             services.AddNavigation();
 
             services.AddScoped<IShapeTableProvider, NavigationShapes>();
             services.AddScoped<IShapeTableProvider, PagerShapesTableProvider>();
             services.AddShapeAttributes<PagerShapes>();
+            services.Configure<PagerOptions>(options => _configuration.GetSection("OrchardCore:OrchardCore_Navigation:PagerOptions").Bind(options));
```

Comment:
Maybe you meant that this is the following that was not working, right?

    _configuration.GetSection("OrchardCore:OrchardCore_Navigation").Bind(opt);

Maybe this works

    _configuration.GetSection("OrchardCore:OrchardCore_Navigation").Get<PagerOptions>();

But there is an overload of `services.Configure<PagerOptions>()` that takes an `IConfiguration` param without having to do a bind.

    services.Configure<PagerOptions>(
        _configuration.GetSection("OrchardCore:OrchardCore_Navigation").GetSection("PagerOptions"));

Reply:
Right! I'll get that changed. much cleaner 
```
services.Configure<PagerOptions>("OrchardCore:OrchardCore_Navigation:PagerOptions", _configuration);
```

Reply:
That did not work.

Reply:
Me too I always have problems to remember all the overloads

No problem will test it tomorrow if you don't fix it in the meantime

Also I prefer to not have a too long key with here 2 `:` separator that it will always have to parse, better I think to provide to it the right section, or as a compromise the last parent section and then use `PagerOptions` as a key.

Reply:
humm, would you rather have it like this

```
_configuration.GetSection("OrchardCore:OrchardCore_Navigation").GetSection("PagerOptions").Bind(options)
```

Reply:
I'm testing them

Just tried the following that works

            services.Configure<PagerOptions>(_configuration
                .GetSection("OrchardCore:OrchardCore_Navigation")
                .GetSection("PagerOptions"));

Also I think that you don't need the `[DefaultValue(10)]` in `PageOptions.cs`.

Reply:
I guess you are right we don’t need the defaultValue attribute since we are not serializing it into json. 

maybe this would even be better? 
```
  services.Configure<PagerOptions>(_configuration
            .GetSection("OrchardCore")
            .GetSection(“Navigation”)
            .GetSection("PagerOptions"));
```

Reply:
Good also to use a variable for the config so that the delegate closure don't have to reference the startup class.


            var configuration = _configuration
                .GetSection("OrchardCore")
                .GetSection("OrchardCore_Navigation")
                .GetSection("PagerOptions");

            services.Configure<PagerOptions>(configuration);

Reply:
Okay. I would probably set it to the code below.

I am out at the moment. I’ll try to do it in the next hour or tomorrow. Unless you want to merge it now, feel free to make this last update and merge it. 
```
var pagerOptions = _configuration
            .GetSection("OrchardCore")
            .GetSection("OrchardCore_Navigation")
            .GetSection("PagerOptions");

        services.Configure<PagerOptions>(pagerOptions);
```

Reply:
Or this one, useful if we will have other options, but the above one is okay

            var configuration = _configuration
                .GetSection("OrchardCore")
                .GetSection("OrchardCore_Navigation");

            services.Configure<PagerOptions>(configuration.GetSection("PagerOptions"));

            // As an example for future options.
            services.Configure<PagerOptions>(configuration.GetSection("OtherOptions"));


Note: Yes the overload taking a string key and a config doesn't work because the key is intended to point to an individual field, not a nested object.

Reply:
No problem will see tomorrow, I let you choose one of my 2 examples.

E.g. what you just suggested, but don't use `var pagerOptions` as it is not a `PagerOptions` but a `IConfigurationSection` which inherit of `IConfiguration`.

So e.g. `var configuration` or maybe `configurationSection` or just `section` as you want.

Reply:
@jtkech I ended up doing this
```
            var navigationConfig = _configuration
                .GetSection("OrchardCore")
                .GetSection("OrchardCore_Navigation");

            services.Configure<PagerOptions>(navigationConfig.GetSection("PagerOptions"));
```
It bugged me until I added the `navigation` in the variable. But everything should be updated now. I also tested the above and it works as expected.

Reply:
Okay will look at it toonight but looks good!